File: 1726-tuple-with-same-product/1726-tuple-with-same-product.py
class Solution:
    def tupleSameProduct(self, nums: List[int]) -> int:
        ans=0
        l1={}
        n= len(nums)
        for i in range(0,n):
            for j in range(i+1,n):
                product = nums[i]* nums[j]
                if product not in l1:
                    l1[product]=0
                else:
                    l1[product]= l1[product]+1
            
                ans += 8 * l1[product]
        return ans

File: 1726-tuple-with-same-product/1726-tuple-with-same-product.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        unordered_map<int, int> mpp;
        int ans = 0;
        for(int i  =0; i<nums.size();i++)
        {
            for(int j = i+1; j<nums.size(); j++)
            {
                int product = nums[i] * nums[j];
                ans += 8 * mpp[product];
                mpp[product]++;
            }
        }
        return ans;
    }
};